### DISCUSSION TITLE
reality的dest的作用是什么啊？

### DISCUSSION THREAD

**Discussion by lhbdhr** [XTLS/Xray-core#2260] [Q&A]
客户端的SNI只要填写一个服务端配置的serverNames其中之一就可以，那服务端的dest配置随便瞎填都可以啊，dest的作用是什么啊？

**Comment by xchawai** [comment]
同问。
为什么dest要填国外网址，reality的dest如果只起跳转作用，理应能填所有网址。

**Reply by RPRX** [reply]
~~为啥总是有人想不通这个问题~~，填国内域名不仅很奇怪，还会产生回国流量容易被封，更别说 GFW 被动检测、主动探测都能识别它
所以就算“只起跳转作用”（你指的是纯转发吧），也不能填国内域名啊，而且并非纯转发，详见下面的说明

**Reply by xchawai** [reply]
感谢讲解！

**Comment by RPRX** [comment]
> 客户端的SNI只要填写一个服务端配置的serverNames其中之一就可以，那服务端的dest配置随便瞎填都可以啊

虽然 REALITY 支持这么玩，但如果 `dest` 实际上没有那个 serverName，会是个比较明显的特征，所以不应滥用这种“隐藏玩法”，这样才长久

> dest的作用是什么啊？

`dest` 是 REALITY 实时模仿的目标，和 REALITY 有交互，不同于 VLESS fallbacks `dest` 的纯转发，只是写法通用

其实当初放出代码后想改名叫 `target`，~~但是各界人士对 Project X 的关注度很高~~，刚放出代码就有部署教程和各种适配了，就没改

**Reply by lhbdhr** [reply]
感谢大佬的回复。
我当时提问的时候还不明白dest的作用，后来研究了很多issue之后大概明白了。
dest偷指定域名的server hello。在tls握手阶段仿冒dest目标回复server hello。
serverNames则是指定一堆可用的sni，客户端指定其中一个，然后xray改写数据包的sni，让墙认为你访问的是白名单网站。
